src/vulkan/vulkan_renderer.cpp

```diff
@@ -296,24 +312,39 @@ namespace IC {
         }
 
         // material descriptors (set 1 or 2)
-        WriteMaterialDescriptors(_allocator, SwapChain::MAX_FRAMES_IN_FLIGHT, writer, *materialData, _textureManager,
+        WriteMaterialDescriptors(_allocator, SwapChain::MAX_FRAMES_IN_FLIGHT, writer, *mesh.Material(), _textureManager,
                                  meshRenderData.materialBuffers);
         for (size_t i = 0; i < SwapChain::MAX_FRAMES_IN_FLIGHT; i++) {
-            int index = materialData->Template().flags & MaterialFlags::Lit ? 2 : 1;
+            int index = mesh.Material()->Template().flags & MaterialFlags::Lit ? 2 : 1;
             writer.UpdateSet(_vulkanDevice.Device(), meshRenderData.descriptorSets[i][index]);
             meshRenderData.UpdateMaterialBuffer(i);
         }
 
         _renderData.push_back(meshRenderData);
     }
 
-    void VulkanRenderer::AddLight(std::shared_ptr<PointLight> light) {
-        _pointLights.push_back(light);
-        AddMesh(light->previewMesh, light->previewMaterial);
+    void VulkanRenderer::AddDirectionalLight(std::shared_ptr<DirectionalLight> &light) {
+        _lightData.directionalLight = light;
     }
 
-    void VulkanRenderer::AddDirectionalLight(std::shared_ptr<DirectionalLight> light) {
-        _directionalLight = light;
+    void VulkanRenderer::AddPointLight(std::shared_ptr<PointLight> &light, std::shared_ptr<Transform> &transform) {
+        if (_lightData.pointLights.size() < MAX_POINT_LIGHTS) {
+            _lightData.pointLights.push_back({light, transform});
+        }
+    }
+
+    void VulkanRenderer::AddGameObject(std::shared_ptr<GameObject> object) {
+        for (auto &component : object->Components()) {
+            if (typeid(*component) == typeid(Mesh)) {
```

Comment:
Make sure `<typeinfo>` is included, otherwise this comparison could be ill-formed.